### PR TITLE
Make replicas optional for the head spec.

### PR DESCRIPTION
**Pull request by DmitriGekhtman**
## Why are these changes needed?

A Ray cluster only supports one head pod.
Accordingly, the KubeRay operator ignores headGroupSpec.Replicas and creates exactly one head pod.

To match this operator behavior and simplify configuration, this PR makes headGroupSpec.Replicas optional and labels the field as deprecated.

We might have anticipated HA or FT features might be implemented via multiple heads, but that's not the route we're going down.

## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by brucez-anyscale — APPROVED** [review]
lgtm. Just make sure the tests pass.

File: ray-operator/apis/ray/v1alpha1/raycluster_types.go [hunk 1]
```diff
@@ -30,9 +30,8 @@ type HeadGroupSpec struct {
 	ServiceType v1.ServiceType `json:"serviceType"`
 	// EnableIngress indicates whether operator should create ingress object for head service or not.
 	EnableIngress *bool `json:"enableIngress,omitempty"`
-	// Number of desired pods in this pod group. This is a pointer to distinguish between explicit
-	// zero and not specified. Defaults to 1.
-	Replicas *int32 `json:"replicas"`
+	// HeadGroupSpec.Replicas is deprecated and ignored; there can only be one head pod per Ray cluster.
+	Replicas *int32 `json:"replicas,omitempty"`
```
Comment (Jeffwan) [inline ray-operator/apis/ray/v1alpha1/raycluster_types.go:34]: We need to make sure all the references are cleaned up. I remember some logics do use it and we can change to use const 1 instead.  Please search the references and have a check.

Comment (DmitriGekhtman) [reply ray-operator/apis/ray/v1alpha1/raycluster_types.go:34]: I actually don't see any references.
The reconcilePods logic definitely doesn't use it -- it just enforces one head.
Grepping for `\.Replicas` indicates it's only accessed on workers (in `util.go` and `raycluster_controller.go`).
vim-go's `:GoReferrers` also doesn't turn up anything (outside of test files)

Comment (DmitriGekhtman) [reply ray-operator/apis/ray/v1alpha1/raycluster_types.go:34]: @Jeffwan do you think you could double-check this?

Comment (Jeffwan) [reply ray-operator/apis/ray/v1alpha1/raycluster_types.go:34]: Let me take a look. If there's nothing left, we can merge the PR.

**Review by Jeffwan — APPROVED** [review]

**Comment by Jeffwan** [comment]
We use Int32Ptr(1) now and doesn't accept any values from this field.

**Pull request merged**